src/astro/databases/snowflake.py

```diff
@@ -57,6 +81,321 @@ def get_table_qualified_name(table: Table) -> str:  # skipcq: PYL-R0201
         qualified_name = ".".join(name for name in qualified_name_lists if name)
         return qualified_name
 
+    def check_optimised_path(self, source_file: File, target_table: Table) -> bool:
+        """
+        Check if there is an optimised path for source to destination.
+
+        :param source_file: File from which we need to transfer data
+        :param target_table: Table that needs to be populated with file data
+        """
+        file_type = self.optimized_path_supported_file_types.get(source_file.type.name)
+        location_type = self.native_support.get(source_file.location.location_type)
+        return bool(file_type and location_type)
+
+    def optimised_transfer(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        **kwargs,
+    ) -> None:
+        """
+        Checks if optimised path for transfer between File location to database exists
+        and if it does, it transfers it and returns true else false.
+
+        :param source_file: Source file
+        :param target_table: Output target table on snowflake
+        :param if_exists: Update strategy for file
+        :param kwargs:
+        """
+        # Fetch snowflake integration passed by the user as part of output table.
+        self.storage_integration = target_table.optional_args.get(
+            "storage_integration", None
+        )
+        if self.storage_integration is None:
+            raise Exception(
+                "To load dataset to snowflake using optimisation, specify the correct storage integration with correct"
+                " permission to the input file. Pass storage_integration={valid storage integration name} as "
+                "optional_args to output_table. "
+                "Read more: https://docs.snowflake.com/en/sql-reference/sql/create-storage-integration.html "
+                "Else disable optimisation by passing use_native_support=False in load_file."
+            )
+        method_name = self.native_support.get(source_file.location.location_type)
+        if method_name:
+            transfer_method = self.__getattribute__(method_name)
+            transfer_method(
+                source_file=source_file,
+                target_table=target_table,
+                if_exists=if_exists,
+                **kwargs,
+            )
+
+    def gs_to_snowflake(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        **kwargs,
```

Comment:
@sunank200  we need to make sure the users can pass other options related to the `copy into` query it might be useful for a lot of common cases.